src/components/Tooltip/Tooltip.js

```diff
@@ -56,18 +55,19 @@ class Tooltip extends PureComponent {
   }
 
   renderTriggerElement() {
-    const { target } = this.props
+    const { target, isVisible } = this.props
     const { show } = this.state
 
     if (!target) { return }
+    const extraProps = isVisible == null ? { onClick: this.handleToggle.bind(this) } : {}
```

Comment:
if a component is controlled I dont think it should also control itself, onClick behavior should work only if isVisible is not passed in.